test/functional/transactions_tests.js

```diff
@@ -156,27 +196,43 @@ describe('Transactions', function() {
   });
 });
 
-function generateTestSuiteTests(testSuites, testContext) {
+function generateTopologyTests(testSuites, testContext) {
+  [
+    { topology: 'replicaset', minServerVersion: '>=3.7.x' },
+    { topology: 'sharded', minServerVersion: '>=4.1.5' }
+  ].forEach(config => {
+    describe(config.topology, function() {
+      generateTestSuiteTests(testSuites, testContext, config);
+    });
+  });
+}
+
+function generateTestSuiteTests(testSuites, testContext, config) {
   testSuites.forEach(testSuite => {
     const minServerVersion = testSuite.minServerVersion
       ? `>=${testSuite.minServerVersion}`
-      : '>=3.7.x';
+      : config.minServerVersion;
+
+    const metadata = {
+      requires: {
+        topology: [config.topology],
```

Comment:
Here `config` is a configuration object described at [line 200](https://github.com/mongodb/node-mongodb-native/pull/1930/files#diff-e012b2e664bc36aa05a0a69c5a7cb81cR200). I believe this will be going away with the new test runner format?